com.alibaba_inc.dingtalk.yml

```diff
@@ -0,0 +1,96 @@
+app-id: com.alibaba_inc.dingtalk
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: dingtalk
+tags: [proprietary]
+separate-locales: false
+
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --filesystem=xdg-run/pipewire-0
+  - --system-talk-name=org.freedesktop.login1
+modules:
+  # To generate icons of specific sizes
+  - name: imagemagick
+    cleanup:
+      - /bin
+      - /etc
+      - /share
+      - /lib
+    config-opts:
+      - --enable-shared
+      - --disable-static
+      - --disable-docs
+      - --with-modules
+      - --with-threads
+    sources:
+      - type: archive
+        url: https://github.com/ImageMagick/ImageMagick/archive/7.1.0-56.tar.gz
+        sha256: 58c36ca7bb7b3d78fee6036945e79e0bb57f67015af08bb58c20488df5158d88
+        x-checker-data:
+          type: anitya
+          project-id: 1372
+          stable-only: true
+          url-template: https://github.com/ImageMagick/ImageMagick/archive/$version.tar.gz
+
+  - name: dingtalk
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 com.alibaba_inc.dingtalk.desktop "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - install -Dm644 com.alibaba_inc.dingtalk.metainfo.xml "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml"
+      - install -Dm755 dingtalk.sh "${FLATPAK_DEST}/bin/dingtalk"
+      - |
+        for res in 16 32 48 64 128 256 512; do
+          convert +set date:create +set date:modify logo.png -resize ${res}x${res} \
+            -background transparent -gravity center -extent ${res}x${res} icon_${res}.png || exit 1
+          install -Dm0644 icon_${res}.png ${FLATPAK_DEST}/share/icons/hicolor/${res}x${res}/apps/${FLATPAK_ID}.png || exit 1
+        done
```

Comment:
IMO you should manually downscale the logo to 512x512 and just commit it to the git repo. No need to create smaller variants, and it will also make imagemagick redundant.